miden-crypto/src/dsa/rpo_falcon512/hash_to_point.rs

```diff
@@ -63,9 +63,21 @@ pub fn hash_to_point_shake256(message: &[u8], nonce: &Nonce) -> Polynomial<Falco
         reader.read(&mut randomness);
         let t = ((randomness[0] as u32) << 8) | (randomness[1] as u32);
         if t < K * MODULUS as u32 {
-            coefficients.push(FalconFelt::new((t % MODULUS as u32) as i16));
+            coefficients.push(u32_to_falcon_felt(t));
         }
```

Comment:
RPO-based implementation seems to be much simpler than this (i.e., we don't check if a random value is smaller than `K * MODULUS`) - is this OK?